<?xml version="1.0"?>
<launch>
  <node pkg="laser_filters" type="scan_to_scan_filter_chain" output="screen" name="laser_filter">
	  <rosparam command="load" file="$(find u_turn)/launch/box_filter.yaml" />
	  <remap from="scan_filtered" to="u_turn_scan_filtered"/>
  </node>
  

  <node pkg="u_turn" name="u_turn" type="u_turn_path.py" output="screen">
	<remap from="raw_obstacles" to="u_turn_raw_obstacles"/>
	  <param name="throttle" value="3" />
	  <param name="max_theta" value="45" />
	  <param name="lateral_offset" type="double" value="4.0" /> <!-- car: 0.6 -->
	  <param name="theta_error_factor" type="double" value="1.2" />
	  <param name="lateral_error_factor" type="double" value="15.0" />
	  <param name="right_steer_scale" type="double" value="1.5" />
	  <param name="left_steer_offset" value="6" />
  </node>

  <node name="obstacle_extractor" pkg="obstacle_detector" type="obstacle_extractor_node">
	<remap from="scan" to="u_turn_scan_filtered"/>
	<remap from="raw_obstacles" to="u_turn_raw_obstacles"/>
    <param name="active"               value="true"/>
    <param name="use_scan"             value="true"/>
    <param name="use_pcl"              value="false"/>

    <param name="use_split_and_merge"    value="true"/>
    <param name="circles_from_visibles"  value="true"/>
    <param name="discard_converted_segments" value="true"/>
    <param name="transform_coordinates"  value="true"/>

    <param name="min_group_points"     value="5"/>

    <param name="max_group_distance"   value="0.1"/>
    <param name="distance_proportion"  value="0.00628"/>
    <param name="max_split_distance"   value="0.2"/>
    <param name="max_merge_separation" value="0.6"/>
    <param name="max_merge_spread"     value="0.2"/>
    <param name="max_circle_radius"    value="0.6"/>
    <param name="radius_enlargement"   value="0.3"/>

    <param name="frame_id"             value="laser"/>
  </node>
</launch>
